### PR TITLE
feat(history_cache): add history_cache

### PR DESCRIPTION
**Pull request by GuillaumeKusiak**

### CODE REVIEW

**Review by MattiasGervilliers — APPROVED** [review]

File: Model/HistoryModel.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
 import { HistoryEntity } from "../Entity/HistoryEntity.js";
+const S_CACHE_NAME = "cache-hothothot";
 
 export class HistoryModel {
   static #O_singleton = null;
 
   #O_historyEntity;
+  #S_CACHE_URL_HISTORY;
```
Comment (MariusPisto) [inline Model/HistoryModel.js]: add const ?

File: Views/html/homepage.html [hunk 2]
```diff
@@ -16,6 +16,8 @@
           .catch(error => console.log('Service Worker registration failed:', error));
         }
     </script>
+    <!-- install pwa button -->
+    <script src="../assets/script/pwaInstall.js" defer></script>
```
Comment (MariusPisto) [inline Views/html/homepage.html:20]: type module ?

**Review by MariusPisto — CHANGES_REQUESTED** [review]
utilise le nommage du prof

File: Model/HistoryModel.js (hunk 1 above)
Comment (GuillaumeKusiak) [reply Model/HistoryModel.js]: const doesn't work

File: Views/html/homepage.html (hunk 2 above)
Comment (GuillaumeKusiak) [reply Views/html/homepage.html:20]: maybe not usefull

File: Model/HistoryModel.js [hunk 3]
```diff
@@ -5,13 +5,12 @@ export class HistoryModel {
   static #O_singleton = null;
 
   #O_historyEntity;
-  #S_CACHE_URL_HISTORY;
+  #S_CACHE_URL_HISTORY = "history/data";
```
Comment (MariusPisto) [inline Model/HistoryModel.js:8]: can you test it pls

**Review by MariusPisto — COMMENTED** [review]
check comments

File: Model/HistoryModel.js (hunk 3 above)
Comment (GuillaumeKusiak) [reply Model/HistoryModel.js:8]: yes no problem, it work !

**Review by MariusPisto — APPROVED** [review]
👍

**Pull request merged**